crates/app-host/src/block_tracer.rs

```diff
@@ -0,0 +1,78 @@
+
+use anyhow::Result;
+
+use scroll_executor::BlockTrace;
+use tokio::runtime::Runtime;
+use std::sync::Arc;
+use l2_client::L2Client;
+
+pub struct BlockTracer {
+    l2_client: Arc<l2_client::L2Client>,
+    rt: Runtime,
+}
+
+impl BlockTracer {
+    pub fn new() -> Self {
+        todo!()
+    }
+
+    pub async fn get_block_traces(&self, blocks: Vec<u64>) -> Result<Vec<BlockTrace>> {
+        let mut handles = Vec::with_capacity(blocks.len());
+
+        for block in blocks {
+            let l2_client = self.l2_client.clone();
+            // todo: use a singleton tokio runtime with limited worker to spawn task
+            let handle = self.rt.spawn(async move {
+                BlockTracer::get_block_trace(l2_client, block).await
+            });
```

Comment:
Why is it necessary that `BlockTracer` has its own runtime? Can't you just use the current runtime?

```suggestion
            // todo: use a singleton tokio runtime with limited worker to spawn task
            let handle = tokio::spawn(async move {
                BlockTracer::get_block_trace(l2_client, block).await
            });
```

Reply:
I want to limit the concurrency of this procedure. Currently the batch is proved sequentially according to batch_index, where there may be tens of blocks within one batch. It may already be useful to have the limit. What's more, in case there is possibility that batches can be proved concurrently, this need not be changed then